File: 240430/방향에 맞춰 이동/move-in-direction.py
x,y=0,0
dx,dy=[1,0,-1,0],[0,-1,0,1]

for _ in range(int(input())):
    direction,num=input().split()
    for _ in range(int(num)):
        if direction=='E':
            x+=dx[0]
        elif direction=='S':
            y+=dy[1]
        elif direction=='W':
            x+=dx[2]
        else:
            y+=dy[3]

print(x,y)

File: 251210/방향에 맞춰 이동/move-in-direction.js
const fs= require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const moves = input.slice(1).map((move)=>move.split(' '));

const dx = [1,0,-1,0];
const dy = [0,-1,0,1];

let x=0, y=0;
for(let move of moves){
    let [d,num]= move;
    num = Number(num);

    let moveDir = 0;
    if(d==='E'){
        moveDir=0;
    }else if(d==='S'){
        moveDir=1;
    }else if(d==='W'){
        moveDir=2;
    }else{
        moveDir=3;
    }
    for(let i=0; i<num; i++){
        x+=dx[moveDir], y+=dy[moveDir];
    }
}

console.log(x,y)